framework/src/main/java/com/jfcbuilder/builders/XYTitleBuilder.java

```diff
@@ -0,0 +1,193 @@
+    this.x = x;
+    return this;
+  }
+
+  
+  public double x() {
+    return x;
+  }
+
+  /**
+   * Sets the y-coordinate of the annotation.
+   * <p>
+   * As of JFreeChart v1.5.3 {@link XYTitleAnnotation} hard-codes the coordinate type to
+   * {@link XYCoordinateType#RELATIVE} therefore the X value being set must be in range (0.0, 1.0). See <a href=
+   * "https://github.com/jfree/jfreechart/blob/v1.5.3/src/main/java/org/jfree/chart/annotations/XYTitleAnnotation.java#L125">v1.5.3
+   * XYTitleAnnotation.java#L125</a>
+   * 
+   * @param y The coordinate value to be set. Should be in range (0.0, 1.0)
+   * @return Reference to this builder instance for method chaining
+   */
+  public XYTitleBuilder y(double y) {
+    this.y = y;
+    return this;
+  }
+
+  public double y() {
+    return y;
+  }
+
+  public XYTitleBuilder maxWidth(double w) {
+    maxWidth = w;
+    return this;
+  }
+
+  public double maxWidth() {
+    return maxWidth;
+  }
+
+  public XYTitleBuilder maxHeight(double h) {
+    maxHeight = h;
+    return this;
+  }
+
+  public double maxHeight() {
+    return maxHeight;
+  }
+
+  public XYTitleBuilder title(Title t) {
+    title = t;
+    return this;
+  }
+
+  public Title title() {
+    return title;
+  }
+
+  public XYTitleBuilder anchor(RectangleAnchor a) {
+    anchor = a;
+    return this;
+  }
+
+  public RectangleAnchor anchor() {
+    return anchor;
+  }
+
+  private void checkBuildPreconditions() throws IllegalStateException {
+
+    if (Double.isNaN(x) || Double.isNaN(y)) {
+      throw new IllegalStateException("X or Y value not set");
+    }
+
+    if (Double.isNaN(maxWidth)) {
+      throw new IllegalStateException("Max width not set");
+    }
+
+    if (Double.isNaN(maxHeight) || Double.isNaN(y)) {
+      throw new IllegalStateException("Max height not set");
+    }
+
+    if (title == null) {
+      throw new IllegalStateException("Title not set");
+    }
+
+    if (anchor == null) {
+      throw new IllegalStateException("Anchor not set");
+    }
+  }
+
+  @Override
+  public XYAnnotation build() throws IllegalStateException {
+
+    checkBuildPreconditions();
+
+    return new XYTitleAnnotation(x, y, title, anchor);
+  }
+
+  /**
+   * Uses {@code Arrays.binarySearch()} to search the source array for the date value. If found, replaces that value in
+   * the builder with the found array index relative to the configured index range. The source time values are assumed
+   * to be timestamps in milliseconds since the epoch start. It's also assumed these are in ascending chronologic order.
+   * Failure to provide them in sorted order will result in undefined behavior as per {@code Arrays.binarySearch()}.
+   */
+  @Override
+  public void mapXToTimeIndex(long[] timeData, int indexRangeStartIndex, int indexRangeEndIndex) {
+    // +1 as the binary search method end index is exclusive.
+    int xIndex = Arrays.binarySearch(timeData, indexRangeStartIndex, indexRangeEndIndex + 1, (long) x());
```

Comment:
It turns out this override logic isn't needed because `XYTitleAnnotation` uses data space coordinates in range (0.0, 1.0). The method should be left blank here instead.